k2/csrc/fsa_algo.h

```diff
@@ -230,23 +230,18 @@ FsaVec LinearFsas(Ragged<int32_t> &symbols);
                  property kFsaPropertiesTopSortedAndAcyclic if you were
                  to compute properties.
 
-   @param [out,optional] entering_arcs   If non-NULL, it will be set to
-                a new Array1, indexed by state_idx01 into `fsas`,
-                saying which arc_idx012 is the best arc entering it,
-                or -1 if there is no such arc.
-
-   @param [out,optional]
-                    If non-NULL, it will contain array of total scores,
-                    of dimension fsas.Dim0(), which will contain the scores
-                    in the final-states of `forward_scores`, or -infinity for
-                    FSAs that had no states.
+   @param [in] entering_arcs  An array indexed by state-index (idx01 into
+                              fsas), i.e.
+                              `entering_arcs.Dim()==fsas.TotSize(1)`,
+                              containing forward scores. (these will be zero
```

Comment:
it should be `containing the best arc entering it (state_idx01)`, not `forward scores`

Reply:
thanks, fixed.